internal/resources/metal/vlan/resource.go

```diff
@@ -104,19 +104,18 @@ func (r *Resource) Create(ctx context.Context, request resource.CreateRequest, r
 }
 
 func (r *Resource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
-	r.Meta.AddFwModuleToMetalUserAgent(ctx, request.ProviderMeta)
-	client := r.Meta.Metal
+	client := r.Meta.NewMetalClientForFramework(ctx, request.ProviderMeta)
 
 	var data ResourceModel
 	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
 	if response.Diagnostics.HasError() {
 		return
 	}
 
-	vlan, _, err := client.ProjectVirtualNetworks.Get(
+	vlan, _, err := client.VLANsApi.GetVirtualNetwork(
```

Comment:
Looks like the missing `href` is in the `facility` object, and there is no way to include `facility.href`, so the API spec will have to be corrected (it says `facility` is an `Href`, but that is impossible).

Reply:
Since facility-based VLANs have long been deprecated, this feels unworthy of a fix in the API spec.  Since `facility.href` is defined as long as `facility` is not in the include list, I opted to instead load the facility details with a separate API request.